File: Search engine/runtime_tests/plugin/logger.py
# -*- coding: utf-8 -*-
import pytest
import logging


pytest_plugins = [
    'runtime_tests.plugin.fs',
]


MAIN_LOG = 'tests.log'
LOG_FORMAT = '%(asctime)s %(levelname)-6s (%(module)s) %(message)s'


@pytest.fixture(scope='session')
def logger(session_fs_manager):
    log_file = session_fs_manager.create_file(MAIN_LOG)
    formatter = logging.Formatter(LOG_FORMAT)
    file_handler = logging.FileHandler(log_file)
    file_handler.setFormatter(formatter)
    session_logger = logging.getLogger()
    session_logger.addHandler(file_handler)
    session_logger.setLevel(logging.INFO)
    return session_logger

File: mail/testopithecus/common/code/utils/logger.ts
import { Registry } from './registry';

export interface Logger {
  log(message: string): void
}

export function log(message: string): void {
  const logger = Registry.get().logger;
  if (logger === null) {
    throw new Error('No logger found!')
  }
  logger.log(message)
}
